src/gfn/gym/helpers/box_utils.py

```diff
@@ -600,8 +601,8 @@ class BoxStateFlowModule(NeuralNet):
     """A deep neural network for the state flow function."""
 
     def __init__(self, logZ_value: torch.Tensor, **kwargs):
-        self.logZ_value = logZ_value
         super().__init__(**kwargs)
+        self.logZ_value = nn.Parameter(logZ_value)
```

Comment:
The only place `BoxStateFlowModule` is used is in `train_box.py`:
```
logZ = torch.tensor(0.0, device=env.device, requires_grad=True)
        # We need a LogStateFlowEstimator

        module = BoxStateFlowModule(
            input_dim=env.preprocessor.output_dim,
            output_dim=1,
            hidden_dim=args.hidden_dim,
            n_hidden_layers=args.n_hidden,
            torso=None,  # We do not tie the parameters of the flow function to PF
            logZ_value=logZ,
        )
```
Naive pytorch question: why do we need `nn.Parameter` ?